### ISSUE TITLE
MA matching Elton John & Britney Spears tracks to Hans Zimmer

### ISSUE THREAD

**Issue by tanderson1992** [music-assistant/support#3334]
### What version of Music Assistant has the issue?

2.3.3

### What version of the Home Assistant Integration have you got installed?

Not installed

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Similar to https://github.com/music-assistant/hass-music-assistant/issues/3141, MA is mismatching filesystem artists when the tracks are properly tagged. After reading 3141, I cleared the DB and scanned again, but still have issues. For some reason, Britney Spears and Elton John performances are being listed as Hans Zimmer. In fact, when I clicked "Refresh Item" while looking at Hans Zimmer, it changed the display name to Elton John and loaded a picture of him. I note that there are still about 160 tracks out of 16,000+ that have improper tags, but none of them are the affected tracks. I am working my way through the 160 tracks and will clear the DB again and report back if that helps.

![image](https://github.com/user-attachments/assets/e4c3dac1-fccd-4dab-b7b0-532737dae37b)
![image](https://github.com/user-attachments/assets/06d2c008-ea37-456b-a097-86f8839af79f)
![image](https://github.com/user-attachments/assets/01d60806-7fd0-4ec8-9d68-6c36dce3475d)
![image](https://github.com/user-attachments/assets/c01f7ecf-b8d7-4941-8638-605707f2ed3a)
![image](https://github.com/user-attachments/assets/b4fe2d2f-735c-4954-8290-29339e49c14c)
(After selecting `Refresh Item` on Hans Zimmer.)

### How to reproduce

Clear the database and wait for re-scan.
Search for artist Hans Zimmer. Note that Britney Spears & Elton John albums are listed under him.
Click Refresh Item. Note that the name and picture changes to Elton John

### Music Providers

Filesystem (remote share) is the only music provider.

### Player Providers

Chromecast, Home Assistant Media Players, and Sonos S1, but this occurs without trying to play the media anywhere.

### Full log output

[music-assistant.log](https://github.com/user-attachments/files/18212723/music-assistant.log)

Note I didn't restart MA after clearing logs, so the new issue starts around line 3625 where I restarted HA, tried to find Hans Zimmer, then cleared the database.

### What version of Home Assistant Core are your running

2024.12.4

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by OzGav** [comment]
All reports of this kind of issue recently has turned out to be either a provider misconfiguration, a track with multiple artists (neither likely in your case) or a tagging problem. 

Once you have cleaned those last 160 do a DB reset and then restart MA and see what happens. If there is still a problem then make a minimal folder with just the affected tracks, turn on debug for the metadata controller, reset the DB and then import that.

**Comment by tanderson1992** [comment]
I cleared up the remaining tracks using Mp3tag (which detected over 400 without albumartist), cleared the database, restarted MA, and waited for the scan to complete. Hans Zimmer is still listed as the artist for most of Elton John and Britney Spears albums. When I go to an album marked as HZ, I can refresh item and it will be changed to EJ with all the tracks now listed as Hans Zimmer and Elton John. If I `Show Info` for each track individually, I can refresh each track to return it to EJ-only one at a time. I don't see a way to refresh all tracks in an album because as I mentioned refreshing the album doesn't remove HZ from the tracks.

Apparently I didn't turn on debug in the correct spot, because turning it on for Filesystem resulted in lots of `(MainThread) [music_assistant.filesystem_smb] Processing: [song path]` with no information about how it was being matched. Before I clear the database and run again against those artists, can you please clarify where I need to turn on debug? Many of the other settings are set to `global (default)` but I can't find where the global log setting is determined after clicking through the setting tabs.

**Comment by OzGav** [comment]
Yes that is possible. There is an open item on the backlog for "Multiple Artists not always being identifed correctly in the database". You can do a minimal share as I said above and test this. Just add a single track or album without the multiple artists and see if that works ok. Then add the one with multiple artists. You may find the problem occurs depending on which you add first.

**Comment by tanderson1992** [comment]
Thanks. Before I do that, can you tell me where in the settings I need to enable debug? Turning it on for Filesystem didn't capture any useful data. It seems like every subsystem has a `global [default]` setting, but I haven't been able to locate where global is defined.

**Comment by OzGav** [comment]
Try MA SETTINGS >> CORE >> METADATA CONTROLLER

**Comment by tanderson1992** [comment]
I haven't been home to test the minimal install yet, but I have been playing with it remotely. I re-scanned all of the Britney Spears songs to properly show them as her and not Hans Zimmer. Then it made her be a collaborator on Gladiator with HZ. (I got no end of chuckling about Russell Crowe in his Gladiator costume singing "Ooh Baby Baby," but that's besides the point.) Anyway I then scanned the Gladiator songs one by one, and after they were all changed back to Hans Zimmer and the proper collaborator (I forget who), Britney Spears was changed back to HZ. Since it seems to be uncorrectable, I've stopped re-scanning the tracks to have them show the proper artist. I also noted the artist names are correct in the playlists even though the individual track info is wrong when you click show information.

I might not be able to setup a minimal share before the issue goes stale again, but I'll report back if I do.

**Comment by OzGav** [comment]
This has been fixed in 2.4.0rc2. Remove all affected files from the library and then resync
